filecoin-proofs/src/api/internal.rs

```diff
@@ -443,16 +444,15 @@ pub fn seal<T: Into<PathBuf> + AsRef<Path>>(
     sector_id_in: &FrSafe,
 ) -> error::Result<SealOutput> {
     let sector_bytes = usize::from(sector_config.sector_bytes());
-    let f_in = File::open(in_path)?;
 
-    // Read all the provided data, even if we will prove less of it because we are faking.
-    let mut data = Vec::with_capacity(sector_bytes);
-    f_in.take(sector_bytes as u64).read_to_end(&mut data)?;
+    // Copy unsealed data to output location, where it will be sealed in place.
+    copy(&in_path, &out_path)?;
+    let f_data = OpenOptions::new().read(true).write(true).open(&out_path)?;
 
-    // Zero-pad the data to the requested size.
-    for _ in data.len()..sector_bytes {
-        data.push(0);
-    }
+    // Zero-pad the data to the requested size by extending the underlying file if needed.
+    f_data.set_len(sector_bytes as u64)?;
+
+    let mut data = unsafe { MmapOptions::new().map_mut(&f_data).unwrap() };
```

Comment:
This might be a good opportunity to start fleshing out a basic `mmap` API that could be leveraged in https://github.com/filecoin-project/rust-fil-proofs/issues/481 (instead of everyone calling `MmapOptions::new()` on its own, which is actually what I'm doing atm).

Reply:
@schomatis Can you clarify what you are doing? Are you calling `MmapOtions::new()` or creating a basic `mmap` API?